scripts/metadata_parser.py

```diff
@@ -120,22 +124,56 @@ def read_new_metadata(filename, module_name, table_name, scheme_name = None, sub
             container = encode_container(module_name, scheme_name, table_name)
         for new_var in new_metadata_header.variable_list():
             standard_name = new_var.get_prop_value('standard_name')
-            rank = len(new_var.get_prop_value('dimensions'))
+            # DH* 2020-05-26
+            # Legacy extension for inconsistent metadata (use of horizontal_dimension versus horizontal_loop_extent).
+            # Since horizontal_dimension and horizontal_loop_extent have the same attributes (otherwise it doesn't
+            # make sense), we swap the standard name and add a note to the long name
+            legacy_note = ''
+            if standard_name == 'horizontal_loop_extent' and scheme_name and \
+                    (table_name.endswith("_init") or table_name.endswith("finalize")):
+                logging.warn("Legacy extension - replacing variable 'horizontal_loop_extent'" + \
+                             " with 'horizontal_dimension' in table {}".format(table_name))
+                standard_name = 'horizontal_dimension'
+                legacy_note = ' replaced by horizontal dimension (legacy extension)'
+            elif standard_name == 'horizontal_dimension' and scheme_name and table_name.endswith("_run"):
```

Comment:
Since this is being handled in the CCPP framework here, do we not need to clean up the scheme metadata? Or should that still be done to avoid future confusion (and to set the right example for future developers)?

Reply:
Correct, this is a temporary workaround - we need to do a full cleanup of the metadata in `GFS_typedefs.F90` and other places. I didn't want to do this as part of this commit, because it would have required a huge amount of changes on top of all that is in the current PRs already. Making this metadata change should be a self-contained PR that does nothing but to remove the workaround and correct the metadata in ccpp-physics and fv3atm.